src/compiler/config/validate-config.ts

```diff
@@ -132,8 +145,11 @@ export const validateConfig = (userConfig?: Config) => {
     return arr;
   }, [] as RegExp[]);
 
+  // this is well justified I promise :)
+  let validatedConfig: Config = config as any;
```

Comment:
this is, afaict, unfortunately necessary for now. The thing to do is to refactor all of this validation so that we do something like:

```ts
const obj = {
  flags: validateFlags(userConfig),
  outputTargets: validateOutputTargets(userConfig),
  ...
}
```

basically 1) refactoring the 'sub-validators' to take an `UnvalidatedConfig` (and whatever else they need) and return the `Config` prop they're concerned with and 2) then building up an object based on that which conforms to the `Config` interface. This can then be returned from the function and we'll have full type-checking of everything going on to convert an `UnvalidatedConfig` to a `ValidatedConfig`.

Reply:
Hmmm with `strictNullChecks: true` set, I'm not seeing any errors if I revert this change (diff below just in case).  Do we still need this change?

```diff
-  // this is well justified I promise :)
-  let validatedConfig: Config = config as any;
-
   return {
-    config: validatedConfig,
+    config,
     diagnostics,
   };
```

Reply:
yeah I was able to delete it without problems - I think I had the `as` cast there when I was experimenting with making some of the properties on `Config` required